### PR TITLE
Register the task first when executing the workflow

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
Failed to execute a workflow when we run `remote.execute(chain_tasks.workflows.example.chain_tasks_wf, inputs={}, version=version, wait=False)`  because we try to register a Luanch Plan before registering tasks

- Register the task first when registering the workflow
- Add `image_config` in Options, so we can set up the image when creating a remote client or executing the workflow
- Remove `from_project` and `from_domain` because we already have `project` and `domain` in args

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: flytekit/configuration/__init__.py [hunk 1]
```diff
@@ -459,6 +459,7 @@ class Config(object):
     secrets: SecretsConfig = SecretsConfig()
     stats: StatsConfig = StatsConfig()
     data_config: DataConfig = DataConfig()
+    image_config: ImageConfig = ImageConfig()
```
Comment (wild-endeavor) [inline flytekit/configuration/__init__.py:462]: I think this was intentionally left out of here, and put into just the `SerializationSettings` object.  We'll have to add it to FlyteRemote separately.

File: flytekit/remote/remote.py [hunk 2]
```diff
@@ -473,7 +475,7 @@ def register_launch_plan(
 
         if serialization_settings is None:
             serialization_settings = SerializationSettings(
-                image_config=None, project=self.default_project, domain=self.default_domain
+                image_config=self._config.image_config, project=self.default_project, domain=self.default_domain
```
Comment (wild-endeavor) [inline flytekit/remote/remote.py:478]: We can leave it None for now... we're only registering a Launch Plan, which we don't need image for, at least for now.

File: flytekit/remote/remote.py [hunk 3]
```diff
@@ -790,7 +779,7 @@ def _(
                     f" image cannot be automatically deducted. Please register and then execute."
                 )
             ss = SerializationSettings(
-                image_config=None,
+                image_config=self._config.image_config,
```
Comment (wild-endeavor) [inline flytekit/remote/remote.py:782]: i'm confused, so this never worked in the past?  oh boy
